test/integration/translation_failures_test.go

```diff
@@ -460,3 +465,115 @@ func validService() *corev1.Service {
 		},
 	}
 }
+
+func TestTranslationFailuresUnsupportedObjectsWithExpressionRoutes(t *testing.T) {
```

Comment:
How about not adding an integration test for this? I think it can be tested just fine on the `Parser.Build` unit level which will be way faster to execute and we can cover all of the unsupported kinds there.

Reply:
Added unit tests and also kept the integration test. Decided not to remove the integration tests because unit tests could not verify that events are attached to k8s objects.

Reply:
Did you consider using the env test for this? If I get it right, you're trying to test creation of events and that should be possible.

Reply:
Do you mean some tests like in `internal/controllers/gateway/httproute_controller_envtest_test.go`? What is the difference between this and running an integration test case? Does this create a separate k8s env, but not deploy kong gateway for each run?

Reply:
> Added unit tests and also kept the integration test. Decided not to remove the integration tests because unit tests could not verify that events are attached to k8s objects.

Hmm, isn't this functionality tested in `TestTranslationFailures` integration test?  I mean - it already verifies that *any* translation failure will result in emitting an event attached to the causing object (after some thinking I would even consider dropping the majority of the test cases defined for it and leave just one that verifies that an arbitrary translation failure happens we get an event with the object associated). 

IMO integration tests should not verify every possible case in which the translation failure might happen as it would be overkill. The mechanism for handling them when they already get out of the parser is common.

Reply:
> Does this create a separate k8s env, but not deploy kong gateway for each run?

@randmonkey Those create a limited k8s environment with kube-apiserver and etcd started locally. You don't get any k8s controllers running but you can create objects and asserts on them or watch for their creation which is basically what you'd want to do in those tests right?

https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest

But @czeslavo also has a point.

One thing for sure: I'd do away with the integration test here as I agree it is an overkill

Reply:
removed integration tests.